src/include/parser/query/reading_clause/in_query_call_clause.h

```diff
@@ -1,25 +1,33 @@
 #pragma once
 
-#include "parser/expression/parsed_expression.h"
+#include "parser/expression/parsed_function_expression.h"
 #include "parser/query/reading_clause/reading_clause.h"
 
 namespace kuzu {
 namespace parser {
 
 class InQueryCallClause : public ReadingClause {
 public:
-    InQueryCallClause(
-        std::string optionName, std::vector<std::unique_ptr<ParsedExpression>> parameters)
-        : ReadingClause{common::ClauseType::IN_QUERY_CALL}, funcName{std::move(optionName)},
-          parameters{std::move(parameters)} {}
+    InQueryCallClause(std::unique_ptr<ParsedExpression> functionExpression)
+        : ReadingClause{common::ClauseType::IN_QUERY_CALL}, functionExpression{
+                                                                std::move(functionExpression)} {}
 
-    inline std::string getFuncName() const { return funcName; }
+    inline std::string getFuncName() const {
```

Comment:
Can we only expose a `ParsedExpression* getFunctionExpression()` API here. And let the caller do casting and get what they need. So that we can only include parsed_expression.h here.